File: android/app/src/main/java/com/reactnativeecommercetask/AuthModule.kt
package com.reactnativeecommercetask

import com.facebook.react.bridge.*
import com.google.firebase.auth.FirebaseAuth

class AuthModule(reactContext: ReactApplicationContext) :
    ReactContextBaseJavaModule(reactContext) {

    private val auth: FirebaseAuth = FirebaseAuth.getInstance()

    override fun getName(): String {
        return "AuthModule"
    }

    @ReactMethod
    fun register(email: String, password: String, promise: Promise) {
        auth.createUserWithEmailAndPassword(email, password)
            .addOnSuccessListener {
                promise.resolve("success")
            }
            .addOnFailureListener { e ->
                promise.reject("REGISTER_ERROR", e.message)
            }
    }

    @ReactMethod
    fun login(email: String, password: String, promise: Promise) {
        auth.signInWithEmailAndPassword(email, password)
            .addOnSuccessListener {
                promise.resolve("success")
            }
            .addOnFailureListener { e ->
                promise.reject("LOGIN_ERROR", e.message)
            }
    }

    @ReactMethod
    fun logout(promise: Promise) {
        auth.signOut()
        promise.resolve("success")
    }
}

File: ios/AuthModule.swift
import Foundation
import FirebaseAuth
import React

@objc(AuthModule)
class AuthModule: NSObject, RCTBridgeModule {
  static func moduleName() -> String! {
    return "AuthModule"
  }

  static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc
  func register(_ email: String,
                password: String,
                resolver resolve: @escaping RCTPromiseResolveBlock,
                rejecter reject: @escaping RCTPromiseRejectBlock) {
    Auth.auth().createUser(withEmail: email, password: password) { result, error in
      if let error = error {
        reject("REGISTER_ERROR", error.localizedDescription, error)
      } else {
        resolve("success")
      }
    }
  }

  @objc
  func login(_ email: String,
             password: String,
             resolver resolve: @escaping RCTPromiseResolveBlock,
             rejecter reject: @escaping RCTPromiseRejectBlock) {
    Auth.auth().signIn(withEmail: email, password: password) { result, error in
      if let error = error {
        reject("LOGIN_ERROR", error.localizedDescription, error)
      } else {
        resolve("success")
      }
    }
  }

  @objc
  func logout(_ resolve: @escaping RCTPromiseResolveBlock,
              rejecter reject: @escaping RCTPromiseRejectBlock) {
    do {
      try Auth.auth().signOut()
      resolve("success")
    } catch let error {
      reject("LOGOUT_ERROR", "Failed to sign out", error)
    }
  }
}
